packages/blockchain-wallet-v4/src/redux/data/activity/reducers.ts

```diff
@@ -41,54 +49,67 @@ export const activityReducer = (
 ): ActivityStateType => {
   switch (action.type) {
     case AT.FETCH_CUSTODIAL_ACTIVITY_FAILURE:
-      return {
-        ...state,
-        [action.payload.product]: {
-          orders: [
-            Remote.Failure(action.payload.error),
-            ...state[action.payload.product].orders
-          ],
-          transactions: {
-            ...state[action.payload.product].transactions,
-            items: [
-              Remote.Failure(action.payload.error),
-              ...state[action.payload.product].transactions.items
-            ]
-          }
-        } as CustodialActivityType
+      const failure: CustodialActivityType = {
+        orders: {
+          ...state[action.payload.product].orders,
+          status: Remote.Failure(action.payload.error)
+        },
+        transactions: {
+          ...state[action.payload.product].transactions,
+          status: Remote.Failure(action.payload.error)
+        }
       }
-    case AT.FETCH_CUSTODIAL_ACTIVITY_LOADING:
       return {
         ...state,
-        [action.payload.product]: {
-          orders: [Remote.Loading, ...state[action.payload.product].orders],
-          transactions: {
-            ...state[action.payload.product].transactions,
-            items: [
-              Remote.Loading,
-              ...state[action.payload.product].transactions.items
-            ]
-          }
-        } as CustodialActivityType
+        [action.payload.product]: failure as CustodialActivityType
       }
+    // ❓
```

Comment:
@schnogz I'd love to hear your thoughts on below 👇

Reply:
https://github.com/blockchain/blockchain-wallet-v4-frontend/pull/2698/commits/4b07a6aab81942e0077e3c37ba17f7bd730cbfbf#diff-079a24ed23a3c2be028c6e1813529bffR27-R30

My solution, I separated the data from the status of that data.